File: ivef-lib/tags/IVEF_0_1_1/ivef-objc/Classes/IVEFMSG_Pong.m

#import "IVEFMSG_Pong.h"


@implementation IVEFMSG_Pong

- (id) init {
    self = [super init];
    if (self != nil) {
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) setHeader:(IVEFHeader *) val {

    [m_header release];
    m_header = val;
    [m_header retain];
}

- (IVEFHeader *) getHeader {

    return m_header;
}

-(void) setBody:(IVEFBody *) val {

    [m_body release];
    m_body = val;
    [m_body retain];
}

- (IVEFBody *) getBody {

    return m_body;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"Header"]) {
                IVEFHeader * val = [attributeDict objectForKey: key];
                [self setHeader: val];
            }
            else if ([key isEqualToString:@"Body"]) {
                IVEFBody * val = [attributeDict objectForKey: key];
                [self setBody: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<MSG_Pong"];
    [xml appendString:@">\n"];
    [xml appendString: [m_header XML] ];
    [xml appendString: [m_body XML] ];
    [xml appendString: @"</MSG_Pong>\n"];
    return xml;
}

-(NSString *) encode: (NSString *) input {

    NSMutableString *str = [[[NSMutableString alloc] initWithString: input] autorelease];

    [str replaceOccurrencesOfString: @"&" withString: "&amp;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @"<" withString: "&lt;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @">" withString: "&gt;") options: nil searchRange: NSMakeRange(0, [str length])];
    [str replaceOccurrencesOfString: @"\"" withString: "&quot;") options: nil searchRange: NSMakeRange(0, [str length])];

    return str;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"MSG_Pong\n"]];
    [str appendString: [m_header stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    [str appendString: [m_body stringValueWithLead: [lead stringByAppendingString: @"    "]] ];
    return str;
}


@end

File: ivef-tools/tags/IVEF_0_1_0/IVEF_0_1_1/ivef-qt/IVEFMSG_Pong.cpp

#include "IVEFMSG_Pong.h"

MSG_Pong::MSG_Pong() {

}

MSG_Pong::MSG_Pong(const MSG_Pong &val) : QObject() {

    m_header = val.m_header;
    m_body = val.m_body;
}

MSG_Pong & MSG_Pong::operator=(const MSG_Pong &val) {

    m_header = val.m_header;
    m_body = val.m_body;
    return *this;
}

QString MSG_Pong::encode( QString str) {

    str.replace('&', "&amp;");
    str.replace('<', "&lt;");
    str.replace('>', "&gt;");
    str.replace('"', "&quot;");
    return str;
}

void MSG_Pong::setHeader(Header val) {

    m_header = val;
}

Header MSG_Pong::getHeader() const {

    return m_header;
}

void MSG_Pong::setBody(Body val) {

    m_body = val;
}

Body MSG_Pong::getBody() const {

    return m_body;
}

QString MSG_Pong::toXML() {

    QString xml = "<MSG_Pong";
    xml.append(">\n");
    xml.append( m_header.toXML() );
    xml.append( m_body.toXML() );
    xml.append( "</MSG_Pong>\n");
    return xml;
}

QString MSG_Pong::toString(QString lead) {

    QString str = lead + "MSG_Pong\n";
    str.append( m_header.toString(lead + "    ") );
    str.append( m_body.toString(lead + "    ") );
    return str;
}

